### ISSUE TITLE
fix(build): empty file when a file has the name of the node package

### ISSUE THREAD

**Issue by davinkevin** [dominique-mueller/angular-package-builder#54]
Hi, 

I've found a very strange behavior when trying to build the `component-companion` which is a very simple project with only 2 files... I've created another project to let you reproduce the bug : https://github.com/davinkevin/bug-name-angular-package-builder

I think there is a problem when we have a project with its name (in the `package.json`) identical to a file of the project (in my case, at the root of the `lib` folder, next to the `index.ts`).

To demonstrate it, I've created a project simply named `alternative` with in the `/src/lib/` folder the two following files : 

*index.ts*

```
export * from './alternative';
```

*alternative.ts*
```ts
export const foo = 'foo';
```

When I run the following command, I have "empty" file for each format 

```
iMac-de-Kevin:/p/t/alternative kevin | master $ jq '.name' package.json; and yarn build; and cat dist/bundles/*.umd.js
"alternative"
yarn run v1.3.2
$ angular-package-builder

Angular Package Builder

  ➜ Configuration
  ➜ Prepare (line breaks, resources)
  ➜ Compile TypeScript into JavaScript (ES2015, ES5)
  ➜ Generate JavaScript bundles (ES2015, ES5, UMD)
  ➜ Compose package

Success! [4.22 seconds]

✨  Done in 4.43s.
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory() :
	typeof define === 'function' && define.amd ? define(factory) :
	(factory());
}(this, (function () { 'use strict';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

})));
```

If I change the name of the project in the `package.json`, for the same command I have the following result : 

```
iMac-de-Kevin:/p/t/alternative kevin | master $ jq '.name' package.json; and yarn build; and cat dist/bundles/*.umd.js
"another-name"
yarn run v1.3.2
$ angular-package-builder

Angular Package Builder

  ➜ Configuration
  ➜ Prepare (line breaks, resources)
  ➜ Compile TypeScript into JavaScript (ES2015, ES5)
  ➜ Generate JavaScript bundles (ES2015, ES5, UMD)
  ➜ Compose package

Success! [4.17 seconds]

✨  Done in 4.38s.
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
	typeof define === 'function' && define.amd ? define(['exports'], factory) :
	(factory((global['another-name'] = {})));
}(this, (function (exports) { 'use strict';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var foo = 'foo';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

exports.foo = foo;

Object.defineProperty(exports, '__esModule', { value: true });

})));
```

FYI : 

```
ng -v

    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/

Angular CLI: 1.6.6
Node: 9.4.0
OS: darwin x64
Angular: 5.2.2
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.6.6
@angular-devkit/build-optimizer: 0.0.42
@angular-devkit/core: 0.0.29
@angular-devkit/schematics: 0.0.52
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.6
@schematics/angular: 0.1.17
typescript: 2.5.3
webpack: 3.10.0
```

Thanks for your help.

/cc @Neonox31 @Yann29 @Yakoust

**Comment by dominique-mueller** [comment]
Yep, I can reproduce the issue. Really weird, though ...

I initially thought this might be a bundling issue, but a bit of research showed that Angular's compiler (ngc) leads to this result, in particular the `flagModuleOutFile` property. I will try to figure out why this actually is an issue (and how to solve it) within the next few days :)

**Comment by davinkevin** [comment]
Thanks a lot 👍

**Comment by dominique-mueller** [comment]
So, this issue cannot really be fixed. We would have to do some magic renaming / moving files, thus introducing potential further issues.

Therefore, I've instead updated the docs to explain this in greater details. In particular, the README now notes that the build process creates new files at the root, and files other than the entry point should be placed in a subfolder (usually named `src`).

So in short, the workaround of renaming / moving the file is the solution here.

**Comment by dominique-mueller** [comment]
If there is any feedback on the documentation, please let me know or feel free to open an improvement PR :)
